test/common/upstream/zone_aware_load_balancer_fuzz_base.h

```diff
@@ -0,0 +1,50 @@
+#include "test/mocks/upstream/priority_set.h"
+
+#include "load_balancer_fuzz_base.h"
+
+namespace Envoy {
+namespace Upstream {
+class ZoneAwareLoadBalancerFuzzBase : public LoadBalancerFuzzBase {
+public:
+  ZoneAwareLoadBalancerFuzzBase(bool need_local_cluster, const std::string& random_bytestring)
+      : LoadBalancerFuzzBase(), random_bytestring_(random_bytestring) {
+    if (need_local_cluster) {
+      local_priority_set_ = std::make_shared<NiceMock<MockPrioritySet>>();
+      local_priority_set_->getMockHostSet(0);
+    }
+    // If you call local_priority_set_.get() and it's not constructed yet, it'll return a nullptr
+  }
+
+  ~ZoneAwareLoadBalancerFuzzBase() {
+    // This deletes the load balancer first. If constructed with a local priority set, load balancer
+    // with reference local priority set on destruction. Since local priority set is in a base
+    // class, it will be initialized second and thus destructed first. Thus, in order to avoid a use
+    // after free, you must delete the load balancer before deleting the priority set.
+    lb_.reset();
+  }
+
+  // These extend base class logic in order to handle local_priority_set_ if applicable.
+  void
+  initializeASingleHostSet(const test::common::upstream::SetupPriorityLevel& setup_priority_level,
+                           const uint8_t priority_level, uint16_t& port);
+  void updateHealthFlagsForAHostSet(const uint64_t host_priority, const uint32_t num_healthy_hosts,
+                                    const uint32_t num_degraded_hosts,
+                                    const uint32_t num_excluded_hosts,
+                                    const std::string random_bytestring);
+
+  void setupZoneAwareLoadBalancingSpecificLogic();
+
+  void addWeightsToHosts();
+
+  // If fuzzing Zone Aware Load Balancers, local priority set will get constructed sometimes. If not
+  // constructed, a local_priority_set_.get() call will return a nullptr.
```

Comment:
Are you observing local -> panic failover in your fuzzer? I think this is one of the key things you'd want to see covered by a fuzzer on zone aware LBs, as it changes the behavior significantly.

Reply:
Hey Harvey, I believe so. I get 63.3 percent line coverage over source/common/upstream/load_balancer_impl.cc.

Reply:
![Screenshot 2020-10-28 at 12 46 00 AM](https://user-images.githubusercontent.com/39203661/97392084-3d369080-18b7-11eb-9769-228b7929367a.png)